lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -742,6 +742,22 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     if (!Ty->hasTypeParameter())
       Ty = Ty->mapTypeOutOfContext();
 
+    // Strip off top level of type sugar (except for type aliases).
+    // We don't want Optional<T> and T? to get different debug types.
+    while (true) {
```

Comment:
The control flow of this loop is a bit hard to follow.

Reply:
But I also don't have a much better idea.

Reply:
I think much of this code goes away altogether if we give up on supporting `-gdwarf-types`